x/avs/keeper/keeper.go

```diff
@@ -87,31 +96,36 @@ func (k Keeper) AVSInfoUpdateWithOperator(ctx sdk.Context, params *OperatorOptPa
 	if err != nil {
 		return errorsmod.Wrap(err, fmt.Sprintf("error occurred when parse acc address from Bech32,the addr is:%s", operatorAddress))
 	}
+
 	if !k.operatorKeeper.IsOperator(ctx, opAccAddr) {
 		return errorsmod.Wrap(delegationtypes.ErrOperatorNotExist, fmt.Sprintf("AVSInfoUpdate: invalid operator address:%s", operatorAddress))
 	}
+
 	avsInfo, err := k.GetAVSInfo(ctx, params.AvsAddress)
 	if err != nil || avsInfo == nil {
 		return errorsmod.Wrap(err, fmt.Sprintf("error occurred when get avs info,this avs address: %s", params.AvsAddress))
 	}
+
 	avs := avsInfo.GetInfo()
 	addresses := avs.OperatorAddress
 
-	if params.Action == RegisterAction && slices.Contains(avs.OperatorAddress, operatorAddress) {
-		return errorsmod.Wrap(types.ErrAlreadyRegistered, fmt.Sprintf("Error: Already registered，operatorAddress %s", operatorAddress))
-	}
-
-	if params.Action == RegisterAction && !slices.Contains(avs.OperatorAddress, operatorAddress) {
+	switch params.Action {
+	case RegisterAction:
+		if slices.Contains(avs.OperatorAddress, operatorAddress) {
```

Comment:
this **avs.OperatorAddress** has already assgined to **addresses** , why not directly use the addresses variable ?

Besides, when using the slices.Contains method, performance may be affected if there are many elements in the slices.

Reply:
I had previously suggested storing both forward and reverse lookups to work around these performance considerations. Maybe we can add that as a TODO for later.

Reply:
fixed

Reply:
this  slices.Contains(avs.OperatorAddress, operatorAddress) has not fixed.